lib/client.js

```diff
@@ -29,12 +29,22 @@ function defaultId () {
 }
 
 function sendPacket (client, packet, cb) {
+  function emitAndCb () {
+    if ('publish' === packet.cmd) {
+      client.emit('publish', packet.topic, packet.payload, packet);
+    }
+    if (cb) {
+      cb();
+    }
+  }
```

Comment:
Using a function here is definitely not ok, as it will cause a perceivable slowdown in the benchmarks (at least in #345). Moreover, this function will be allocated even for every other packet (where there is no need).